src/EnergyPlus/PlantUtilities.cc

```diff
@@ -915,12 +915,6 @@ void PullCompInterconnectTrigger(EnergyPlusData &state,
         // First store the current check in a single variable instead of array for readability
         CurCriteria = state.dataPlantUtilities->CriteriaChecks(UniqueCriteriaCheckIndex);
 
-        // Check to make sure we didn't reuse the index in multiple components
-        if (CurCriteria.CallingCompLoopNum != plantLoc.loopNum || CurCriteria.CallingCompLoopSideNum != plantLoc.loopSideNum ||
```

Comment:
Would it be useful to change this into an `assert`?

Reply:
I'm not sure,. I think this is the type of thing that should only be done once in the simulation, just not sure where to do that. Maybe move this code up into the `if (UniqueCriteriaCheckIndex <= 0)` block? I would think those 4 Nums would always be non-zero but this code would be better served there than here (and with the assert).

Reply:
The other thing that could be improved is that BranchNum and CompNum are not used. Those could be removed.